codegen/lib/objc-codegen.js

```diff
@@ -0,0 +1,280 @@
+    apiName += '.destroyAll';
+  } else {
+    apiName += '.' + op;
+  }
+
+  // Names of resources/models in Angular start with a capital letter
+  var ngModelName = modelName[0].toUpperCase() + modelName.slice(1);
+  method.internal = 'Use ' + ngModelName + '.' + apiName + '() instead.';
+
+  // build a reverse record to be used in ngResource
+  // Product.__find__categories -> Category.::find::product::categories
+  var reverseName = '::' + op + '::' + modelName + '::' + scopeName;
+
+  var reverseMethod = Object.create(method);
+  reverseMethod.name = reverseName;
+  reverseMethod.internal = 'Use ' + ngModelName + '.' + apiName + '() instead.';
+  // override possibly inherited values
+  reverseMethod.deprecated = false;
+
+  var reverseModel = findModelByName(models, targetClass);
+  reverseModel.methods.push(reverseMethod);
+  if(reverseMethod.name.match(/create/)){
+    var createMany = Object.create(reverseMethod);
+    createMany.name = createMany.name.replace(
+      /create/,
+      'createMany'
+    );
+    createMany.internal = createMany.internal.replace(
+      /create/,
+      'createMany'
+    );
+    createMany.isReturningArray = function() { return true; };
+    reverseModel.methods.push(createMany);
+  }
+
+  var scopeMethod = Object.create(method);
+  scopeMethod.name = reverseName;
+  // override possibly inherited values
+  scopeMethod.deprecated = false;
+  scopeMethod.internal = false;
+  modelClass.scopes[scopeName].methods[apiName] = scopeMethod;
+  if(scopeMethod.name.match(/create/)){
+    var scopeCreateMany = Object.create(scopeMethod);
+    scopeCreateMany.name = scopeCreateMany.name.replace(
+      /create/,
+      'createMany'
+    );
+    scopeCreateMany.isReturningArray = function() { return true; };
+    apiName = apiName.replace(/create/, 'createMany');
+    modelClass.scopes[scopeName].methods[apiName] = scopeCreateMany;
+  }
+}
+
+function findModelByName(models, name) {
+  for (var n in models) {
+    if (n.toLowerCase() == name.toLowerCase())
+      return models[n];
+  }
+}
+
+function addObjCNames(models, modelPrefix) {
+  for (var modelName in models) {
+    var meta = models[modelName];
+    meta.objcModelName = modelPrefix + modelName[0].toUpperCase() + modelName.slice(1);
+    if (meta.baseModel === 'Model' || meta.baseModel === 'PersistedModel') {
+      meta.objcBaseModel = 'LB' + meta.baseModel;
+    } else {
+      console.error('unknown baseModel name: ' + meta.baseModel); // FIXME
+    }
+
+    meta.objcParams = [];
+    for (var param in meta.params) {
+      var type = meta.params[param].type.name;
+      console.log('param type: ' + meta.params[param].type.name); // FIXME
+      if (type === 'String') {
+        meta.params[param].type.objcName = '(nonatomic, copy) NSString *';
+      } else if (type === 'Number') {
+        meta.params[param].type.objcName = 'long ';
+      } else if (type === 'Boolean') {
+        meta.params[param].type.objcName = 'BOOL ';
```

Comment:
I'm not handling Date, Buffer and GeoPoint yet. I'm currently not sure to which class these data types will be converted (or actually successfully concerted) in iOS SDK...